File: Problemset/defanging-an-ip-address/defanging-an-ip-address.go

// @Title: IP 地址无效化 (Defanging an IP Address)
// @Author: KivenC
// @Date: 2019-07-08 13:47:38
// @Runtime: 0 ms
// @Memory: 1.9 MB

func defangIPaddr(address string) string {
    new_addr := []rune{}
    for _, v := range address {
        if v == '.' {
            new_addr = append(new_addr, '[', '.', ']')
        } else {
            new_addr = append(new_addr, v)
        }
    }
    return string(new_addr)
}

File: Problemset/defanging-an-ip-address/defanging-an-ip-address.py

# @Title: IP 地址无效化 (Defanging an IP Address)
# @Author: KivenC
# @Date: 2019-07-08 11:42:09
# @Runtime: 52 ms
# @Memory: 12.9 MB

class Solution:
    def defangIPaddr(self, address: str) -> str:
        return address.replace('.', '[.]')
